### PR TITLE
[WasmFS] Support utime()

### PR DESCRIPTION
**Pull request by kripken**
This implements the `__syscall_utimensat` syscall and everything necessary
for that to get a basic test of `utime` passing.

`getParsedPath()` now receives an optional FD, which if given is the "base"
relative to which the path is parsed. That is what all the at* syscalls will
use.

This also switches us to use `utime.c` from musl instead of from JS. This is
necessary for WasmFS since the JS version only supported the JS FS. But
it works ok in the JS FS as well, as it also implements the relevant syscall,
so this is better anyhow.

(last paragaph was edited, after removing the new stdlib and using utime.c
in all modes)

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -1431,6 +1431,21 @@ def can_use(self):
     return settings.WASMFS
 
 
+# Libc components of WasmFS, which uses more libc code than the JS filesystem.
+# This must be a separate library than libwasmfs because the build commands
+# are different for libc code.
```
Comment (sbc100) [inline tools/system_libs.py:1436]: I hope this isn't necessary.   

There are a couple of ways we can get the correct build flags:

1. Have libwasmfs inherit from MuslInternalLibrary.
2. Use the new `customize_build_cmd` to use different cflags for utime.c.

Comment (kripken) [reply tools/system_libs.py]: Ok, switched to a single library and picking the build flags in `customize_build_cmd`.

File: tools/system_libs.py [hunk 2]
```diff
@@ -1416,17 +1416,32 @@ def can_build(self):
...
                    'js_file_backend.cpp', 'proxied_file_backend.cpp',
                    'fetch_backend.cpp',
                    'js_api.cpp'])
 
+    # The JS filesystem implements e.g. utime() directly in JS, while WasmFS
+    # wants to use the normal musl code that does a syscall, and WasmFS
+    # implements the syscall.
+    libc_files = files_in_path(
+        path='system/lib/libc/musl/src/time',
+        filenames=['utime.c'])
```
Comment (sbc100) [inline tools/system_libs.py:1433]: Any reason we can't just switch to including this file across the board?  (I can take a look as a followup if you like).

File: src/library.js [hunk 3]
```diff
@@ -96,23 +96,6 @@ LibraryManager.library = {
...
-    if (times) {
-      // NOTE: We don't keep track of access timestamps.
-      time = {{{ makeGetValue('times', C_STRUCTS.utimbuf.modtime, 'i32') }}} * 1000;
-    } else {
-      time = Date.now();
-    }
-    return setFileTime(path, time);
-  },
-
   utimes__deps: ['$setFileTime'],
```
Comment (sbc100) [inline src/library.js:99]: I wonder if this can be removed too (perhaps as a followup).

**Pull request merged**